test/components/form/row.spec.js

```diff
@@ -1,80 +1,50 @@
-import { RouterLinkStub } from '@vue/test-utils';
-
 import DateTime from '../../../src/components/date-time.vue';
 import EnketoFill from '../../../src/components/enketo/fill.vue';
 import EnketoPreview from '../../../src/components/enketo/preview.vue';
 import FormRow from '../../../src/components/form/row.vue';
-import FormTable from '../../../src/components/form/table.vue';
 import LinkIfCan from '../../../src/components/link-if-can.vue';
 
+import Form from '../../../src/presenters/form';
+
 import testData from '../../data';
 import { mockLogin } from '../../util/session';
 import { mockRouter } from '../../util/router';
 import { mount } from '../../util/lifecycle';
 
-const mountComponent = () => {
-  // Mounting FormTable rather than FormRow in order not to mock the `columns`
-  // prop of FormRow.
-  const table = mount(FormTable, {
-    container: {
-      router: mockRouter('/projects/1'),
-      requestData: {
-        project: testData.extendedProjects.last(),
-        forms: testData.extendedForms.sorted()
-      }
-    }
-  });
-  return table.getComponent(FormRow);
-};
+const mountComponent = () => mount(FormRow, {
+  props: { form: new Form(testData.extendedForms.last()) },
+  container: {
+    requestData: {
+      forms: testData.extendedForms.sorted(),
+      project: testData.extendedProjects.last()
+    },
+    router: mockRouter('/projects/1')
+  }
+});
 
 describe('FormRow', () => {
   describe('form name', () => {
     beforeEach(mockLogin);
 
     it("shows the form's name if it has one", () => {
-      testData.extendedForms.createPast(1, { xmlFormId: 'f', name: 'My Form' });
+      testData.extendedForms.createPast(1, { xmlFormId: 'f', name: 'My Form', state: 'open' });
       mountComponent().get('.name a').text().should.equal('My Form');
     });
 
     it('shows the xmlFormId if the form does not have a name', () => {
-      testData.extendedForms.createPast(1, { xmlFormId: 'f', name: null });
+      testData.extendedForms.createPast(1, { xmlFormId: 'f', name: null, state: 'open' });
```

Comment:
state not needed here either because the only place form row cares about state is in the actions column